packages/subgraph/schema.graphql

```diff
@@ -8,29 +8,105 @@ interface Token {
   symbol: String
 }
 
-type ERC20Token implements Token @entity(immutable: true) {
+type ERC20Contract implements Token @entity(immutable: true) {
   id: ID! # use address as id
   name: String
   symbol: String
   decimals: BigInt
 }
 
-type ERC721Token implements Token @entity(immutable: true) {
+type ERC721Contract implements Token @entity(immutable: true) {
   id: ID! # use address as id
   name: String
   symbol: String
-  baseURI: String
 }
 
-type Balance @entity {
+# Token Balances
+interface TokenBalance {
   id: ID! # token address + dao address
-  token: ERC20Token! # address(0) corresponds to the native token
+  dao: Dao!
+  lastUpdated: BigInt!
+}
+
+type ERC20Balance implements TokenBalance @entity {
+  id: ID! # token address + dao address
+  token: ERC20Contract! # address(0) corresponds to the native token
   dao: Dao!
   balance: BigInt!
   lastUpdated: BigInt!
 }
 
-# Dao
+type ETHBalance implements TokenBalance @entity {
```

Comment:
Better use `NativeBalance` as we will also be on Polygon and other chains/layers where the native token is not ETH

Reply:
done https://github.com/aragon/core/pull/251/commits/72d501bfd5d066a003a534043150f5f6667b75b4